lib/iris/cube.py

```diff
@@ -880,8 +880,11 @@ def convert_units(self, unit):
 
         """
         # If the cube has units convert the data.
-        if not self.units.is_unknown():
-            self.data = self.units.convert(self.data, unit)
+        if self.units.is_unknown():
+            raise iris.exceptions.UnitConversionError(
+                'cannot convert from unknown units to {!s}, '
+                'set the "units" attribute instead'.format(unit))
```

Comment:
I guess it depends on why the error came up.  If a user knew the unit was unknown and deliberately used the method to set it (as you suggested in your first comment @ajdawson) then telling them the right way to do it makes sense.  If it came up because the data file was dodgy and the user didn't notice they weren't converting from anything (which has happened to me, and prompted posting the issue to begin with) probably all you need is to flag up that the cube unit is unknown.